Sources/MongoSwift/MongoCollection+Write.swift

```diff
@@ -235,48 +193,35 @@ extension MongoCollection {
      */
     @discardableResult
     public func deleteMany(_ filter: Document, options: DeleteOptions? = nil) throws -> DeleteResult? {
-        let opts = try self.encoder.encode(options)
-        let reply = Document()
-        var error = bson_error_t()
-        guard mongoc_collection_delete_many(self._collection, filter.data, opts?.data, reply.data, &error) else {
-            throw getErrorFromReply(bsonError: error, from: reply)
+        do {
+            let model = DeleteManyModel(filter, collation: options?.collation)
+            let result = try self.bulkWrite([model], options: options?.asBulkWriteOpts())
+            return try DeleteResult(from: result)
+        } catch {
+            throw convertBulkWriteError(error)
         }
-
-        guard isAcknowledged(options?.writeConcern) else {
-            return nil
-        }
-
-        return try self.decoder.internalDecode(
-                DeleteResult.self,
-                from: reply,
-                withError: "Couldn't understand response from the server.")
     }
+}
 
-    /**
-     * Returns whether the operation's write concern is acknowledged. If `nil`,
-     * the collection's write concern will be considered.
-     *
-     * - Parameters:
-     *   - writeConcern: `WriteConcern` from the operation's options struct
-     *
-     * - Returns: Whether the operation will use an acknowledged write concern
-     */
-    fileprivate func isAcknowledged(_ writeConcern: WriteConcern?) -> Bool {
-        /* If the collection's write concern is also `nil` it is the default. We
-         * can safely assume it is acknowledged, since the server requires that
-         * getLastErrorDefaults is acknowledged by at least one member. */
-        guard let wc = writeConcern ?? self.writeConcern else {
-            return true
-        }
+/// Protocol indicating that an options type can be converted to a BulkWriteOptions.
+private protocol BulkWriteOptionsConvertible {
+    var bypassDocumentValidation: Bool? { get }
+    var writeConcern: WriteConcern? { get }
+    func asBulkWriteOpts() -> BulkWriteOptions
```

Comment:
nit: Is there benefit to dropping the `ion` here? Consider calling it `asBulkWriteOptions`

Reply:
nah, renamed.